prow/plugins/plugins.go

```diff
@@ -514,6 +517,20 @@ func (pa *PluginAgent) Config() *Configuration {
 	return pa.configuration
 }
 
+// validateConfig ensure that the configuration
+// doesn't contain any nil struct
+func ensureConfig(config *Configuration) error {
```

Comment:
Instead of just ensuring that nothing is `nil`, you should look at `config.Plugins` and ensure that any plugin that is actually in use by a repo somewhere has a non-`nil` configuration struct. It's possible and totally OK if a plugin's configuration is not given when the plugin is not in use.